src/extract/typed_header.rs

```diff
@@ -79,12 +79,14 @@ impl<T> Deref for TypedHeader<T> {
 #[cfg_attr(docsrs, doc(cfg(feature = "headers")))]
 #[derive(Debug)]
 pub struct TypedHeaderRejection {
```

Comment:
I think we should make this `#[non_exhaustive]` so users cannot construct this from the outside. Feels the safest to me and we can always change it if someone has a use-case.